core/src/main/java/org/infinispan/container/versioning/RankCalculator.java

```diff
@@ -0,0 +1,63 @@
+package org.infinispan.container.versioning;
+
+import java.util.List;
+import java.util.concurrent.atomic.AtomicLong;
+
+import org.infinispan.factories.annotations.Inject;
+import org.infinispan.factories.annotations.Start;
+import org.infinispan.factories.scopes.Scope;
+import org.infinispan.factories.scopes.Scopes;
+import org.infinispan.notifications.Listener;
+import org.infinispan.notifications.cachemanagerlistener.CacheManagerNotifier;
+import org.infinispan.notifications.cachemanagerlistener.annotation.ViewChanged;
+import org.infinispan.notifications.cachemanagerlistener.event.ViewChangedEvent;
+import org.infinispan.remoting.transport.Address;
+import org.infinispan.remoting.transport.Transport;
+import org.infinispan.util.logging.Log;
+import org.infinispan.util.logging.LogFactory;
+
+/**
+ * Compute the version prefix to be used by {@link NumericVersionGenerator} in clustered caches.
+ *
+ * @since 14.0
+ */
+@Scope(Scopes.GLOBAL)
+@Listener
+public class RankCalculator {
+   private static final Log log = LogFactory.getLog(org.infinispan.container.versioning.RankCalculator.class);
```

Comment:
nitpic 
```suggestion
   private static final Log log = LogFactory.getLog(MethodHandles.lookup().lookupClass());
```

Reply:
Personally I still like the old comfy class name approach and relying on the IDE to do the rename, but I agree the full name does look kind of ugly.